packages/cli/src/sso/saml/samlHelpers.ts

```diff
@@ -73,39 +76,18 @@ export const isSamlPreferences = (candidate: unknown): candidate is SamlPreferen
 	);
 };
 
-export function generatePassword(): string {
-	const length = 18;
-	const charset = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
-	const charsetNoNumbers = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
-	const randomNumber = Math.floor(Math.random() * 10);
-	const randomUpper = charset.charAt(Math.floor(Math.random() * charsetNoNumbers.length));
-	const randomNumberPosition = Math.floor(Math.random() * length);
-	const randomUpperPosition = Math.floor(Math.random() * length);
-	let password = '';
-	for (let i = 0, n = charset.length; i < length; ++i) {
-		password += charset.charAt(Math.floor(Math.random() * n));
-	}
-	password =
-		password.substring(0, randomNumberPosition) +
-		randomNumber.toString() +
-		password.substring(randomNumberPosition);
-	password =
-		password.substring(0, randomUpperPosition) +
-		randomUpper +
-		password.substring(randomUpperPosition);
-	return password;
-}
-
 export async function createUserFromSamlAttributes(attributes: SamlUserAttributes): Promise<User> {
-	return await Container.get(UserRepository).manager.transaction(async (trx) => {
-		const { user } = await Container.get(UserRepository).createUserWithProject(
+	const randomPassword = randomString(18);
```

Comment:
I don't think it's an issue, but just wanted to point out that the previous function created a password that abided by our password rules, number, upper case, yada yada yada. The new one does not anymore.

Reply:
users created via ldap or saml users can't login with the their default password since that string is ephemeral. there is no point in trying to enforce the password policy here since this generated password can never be used.